pkg/handlers/embedded_cluster_artifacts.go

```diff
@@ -114,3 +113,107 @@ func (h *Handler) GetEmbeddedClusterBinary(w http.ResponseWriter, r *http.Reques
 		return
 	}
 }
+
+// GetEmbeddedClusterInfraImages returns the infrastructure images as a .tgz file
+func (h *Handler) GetEmbeddedClusterInfraImages(w http.ResponseWriter, r *http.Request) {
+	if !util.IsEmbeddedCluster() {
+		logger.Error(errors.New("not an embedded cluster"))
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	// Get k0s directory path from env var
+	k0sDir := util.EmbeddedClusterK0sDir()
+	if k0sDir == "" {
+		logger.Error(errors.New("environment variable EMBEDDED_CLUSTER_K0S_DIR not set"))
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	// Path to images directory
+	imagesDir := filepath.Join(k0sDir, "images")
+
+	// Check if images directory exists
+	_, err := os.Stat(imagesDir)
+	if os.IsNotExist(err) {
+		logger.Error(errors.Wrap(err, "images directory not found"))
+		w.WriteHeader(http.StatusNotFound)
+		return
+	} else if err != nil {
+		logger.Error(errors.Wrap(err, "failed to stat images directory"))
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	// Find all infra image files
+	entries, err := os.ReadDir(imagesDir)
+	if err != nil {
+		logger.Error(errors.Wrap(err, "failed to read images directory"))
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	// Filter and sort image files
+	var infraImages []os.FileInfo
+	for _, entry := range entries {
+		if entry.IsDir() {
+			continue
+		}
+
+		// Check if it's an infra images file
+		// Note: The directory can contain two types of image files:
```

Comment:
Why use mod time and not hardcode to the one with the ec prefix? I'm still not sure I understand the logic here

Reply:
was trying to make it a bit resilient to future name changes in ec, but probably didn't have to. i can make only check for these two names for now

Reply:
updated